src/ripple/overlay/impl/ProtocolVersion.cpp

```diff
@@ -0,0 +1,140 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2019 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/overlay/impl/ProtocolVersion.h>
+#include <ripple/beast/core/LexicalCast.h>
+#include <ripple/beast/rfc2616.h>
+#include <boost/regex.hpp>
+#include <algorithm>
+
+namespace ripple {
+
+/** The list of protocol versions we speak and we prefer to use. */
+constexpr
+ProtocolVersion const supportedProtocolList[]
+{
+    { 1, 2 },
+    { 2, 0 }
+};
+
+constexpr
+ProtocolVersion const legacyProtocol { 1, 2 };
+
+std::string
+to_string(ProtocolVersion const& p)
+{
+    if (p == legacyProtocol)
+        return "RTXP/" + std::to_string(p.first) + "." + std::to_string(p.second);
+
+    return "XRPL/" + std::to_string(p.first) + "." + std::to_string(p.second);
+}
+
+std::vector<ProtocolVersion>
+parseProtocolVersions(boost::beast::string_view const& value)
```

Comment:
Document that this function must return a sorted list (or `negotiateProtocolVersion` breaks)

Reply:
We already do, in the header file, but I've made the wording more explicit and added a comment here.